File: Frontend/caretag/lib/Modules/auth/data/auth/signup_request.dart
class SignUpRequest {
  String? username;
  String? password;
  String? email;

  SignUpRequest({this.username, this.password, this.email});

  SignUpRequest.fromJson(Map<String, dynamic> json) {
    username = json['username'];
    password = json['password'];
    email = json['email'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['username'] = username;
    data['password'] = password;
    data['email'] = email;
    return data;
  }
}

File: Backend/CareTag/src/main/java/com/example/CareTag/DTOs/hospitalDTOs/Signuprequest.java
package com.example.CareTag.DTOs.hospitalDTOs;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Signuprequest {

  private String name;
  private String email;
  private String password;
  private String phone;
  private String website;

  private String address;
  private String city;
  private String state;
  private String pincode;

  private Double latitude;
  private Double longitude;

  private String logoUrl;
  private Integer numberOfBeds;
  private List<String> departments;

  private String registrationNumber;
  private String registrationCertificateUrl;

  private String accreditationNumber;
  private String accreditationType;

  private String clinicalEstablishmentLicenseUrl;
  private String gstNumber;

  private List<BranchDTO> branches;

}
